guides/keras_nlp/neural_machine_translation.py

```diff
@@ -0,0 +1,509 @@
+    # input sequence.
+    def token_probability_fn(decoder_input_tokens):
+        return transformer([encoder_input_tokens, decoder_input_tokens])[:, -1, :]
+
+    # Set the prompt to the "[START]" token.
+    prompt = tf.expand_dims(tf.repeat(spa_tokenizer("[START]")[0], batch_size), axis=1)
+
+    generated_tokens = keras_nlp.utils.greedy_search(
+        token_probability_fn,
+        prompt,
+        max_length=40,
+        end_token_id=spa_tokenizer("[END]")[0],
+    )
+    generated_sentences = spa_tokenizer.detokenize(generated_tokens)
+    return generated_sentences
+
+
+test_eng_texts = [pair[0] for pair in test_pairs]
+for i in range(10):
+    input_sentence = random.choice(test_eng_texts)
+    translated = decode_sequences(tf.constant([input_sentence]))
+    translated = translated.numpy()[0].decode("utf-8")
+    translated = translated.replace("[PAD]", "").strip()
+    print(f"*** Example {i} ***")
+    print(input_sentence)
+    print(translated)
+    print()
+
+"""
+*** Example 0 ***
+have you seen this?
+[START] ¿has visto esto? [END]
+
+*** Example 1 ***
+it's very hot here.
+[START] hace mucho calor aqui. [END]
+
+*** Example 2 ***
+my mother always says she's going to visit me soon.
+[START] mi madre es algo de decir que ella me va a visitar pronto. [END]
+
+*** Example 3 ***
+you can't say that.
+[START] no puedes decir eso. [END]
+
+*** Example 4 ***
+there are always some chores to be done around the house.
+[START] siempre hay algunos chance para hacer cerca de la casa. [END]
+
+*** Example 5 ***
+sometimes the boys would play a joke on the teacher.
+[START] a veces los ninos tocaria una broma con el profesor. [END]
+
+*** Example 6 ***
+move this table toward the corner.
+[START] muevile esta mesa hacia la car, se muelvale esta mesa. [END]
+
+*** Example 7 ***
+tom has never heard mary sing.
+[START] tom nunca ha oído cantar a mary. [END]
+
+*** Example 8 ***
+tom was in town monday night.
+[START] tom estaba en la noche el lunes por la noche. [END]
+
+*** Example 9 ***
+what's happened, has happened. it's history.
+[START] lo que paso, ha pasado. es historia. [END]
+"""
+
+"""
+## Evaluating our Model (Quantitative Analysis)
+
+There are many metrics which are used for text generation tasks. Here, to
+evaluate translations generated by our model, let's compute the ROUGE-2 score.
+Essentially, ROUGE-2 is a score based on the number of common bigrams between
+the reference text and the generated text. We will calculate the score over 30
+test samples (since decoding is an expensive process).
+"""
+
+rouge_n = keras_nlp.metrics.RougeN(order=2)
+
+for test_pair in test_pairs[:30]:
+    input_sentence = test_pair[0]
+    reference_sentence = test_pair[1]
+
+    translated_sentence = decode_sequences(tf.constant([input_sentence]))
+    translated_sentence = translated_sentence.numpy()[0].decode("utf-8")
+    translated_sentence = translated_sentence.replace("[PAD]", "").strip()
```

Comment:
should we be stripping the [START] and [END] as well? or is that in the reference too?